marketplace-service/src/main/java/com/axonivy/market/util/MetadataReaderUtils.java

```diff
@@ -0,0 +1,74 @@
+package com.axonivy.market.util;
+
+import com.axonivy.market.constants.MavenConstants;
+import com.axonivy.market.entity.Metadata;
+import lombok.extern.log4j.Log4j2;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.StringReader;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
+@Log4j2
+public class MetadataReaderUtils {
+  private MetadataReaderUtils() {
+  }
+
+  public static void parseMetadataFromString(String xmlData, Metadata metadata) {
```

Comment:
this method start as parse, then it should return a value, not void.
Also it will make below code is makes sense:
      MetadataReaderUtils.parseMetadataFromString(metadataContent, metadata); => modify metadata
      updateMavenArtifactVersionFromMetadata(artifactVersionCache, metadata); => use the returned metadata from above method parseMetadataFromString
      updateContentsFromNonMatchVersions(releasedVersions, metadata); => use the returned metadata from above method parseMetadataFromString